### DISCUSSION TITLE
Will setting same value to state trigger a rerender?

### DISCUSSION THREAD

**Discussion by Toshinaki** [pmndrs/zustand#1905] [Q&A]
I'm using zustand with immer.
If I defined a function like this:
```
toggle: (open?: boolean) =>
  set((state) => {
    state.open = open === undefined ? !state.open : open;
  }),
```
and use the state inside a component like this:
```
const open = useStore(state => state.open);

return open ? ... : ...
```

if `state.open` is `true`, will `toggle(true)` triggers a rerender?

Should I add a check:
```
toggle: (open?: boolean) =>
  set((state) => {
    if (open !== state.open) {
      state.open = open === undefined ? !state.open : open;
    }
  }),
```

**Comment by dbritto-dev** [comment]
@Toshinaki on `react` each time you call `setState` triggers a new render to optimize re-renders on a child you could use `memo` or you could optimize how you are calling `setState` like only calls `setState` when the value changes.

BTW, you shouldn't worry about that at least you are running slow/heavy operations
